spec.html

```diff
@@ -26016,7 +26016,7 @@ <h1>_NativeError_ Object Structure</h1>
 
       <!-- es6num="19.5.6.1" -->
       <emu-clause id="sec-nativeerror-constructors">
-        <h1>_NativeError_ Constructors</h1>
+        <h1>The _NativeError_ Constructors</h1>
```

Comment:
I'm not sure this is a good change; it reads like proper English to me in master, but reads a bit awkwardly with the "the" (since "constructors" is plural).

Reply:
Well, there's already "Properties of the _NativeError_ Constructors", "The _TypedArray_ Constructors", and "Properties of the _TypedArray_ Constructors" as clause-titles elsewhere, so "The _NativeError_ Constructors" would be consistent with those.

(As for "the <plural>" reading awkwardly, consider "The Merry Wives of Windsor", "The Lovely Bones", and "release the hounds".)

Reply:
Consistency wins, obviously, but personally I’d be more inclined to remove the other definite articles.

@bterlson, @bmeck, thoughts?

Reply:
"The" as a definite article in English works fine with plurals. I think removing it is fine however, since it only is definite per Realm so there is some falsehood to using it here. My main concern is that as more cross realm and multiple realm APIs start to show up in the JS spec that we might confuse people by using the word "the". That said. I think we should document our choice. I'm for removal.

Reply:
>  it only is definite per Realm so there is some falsehood to using it here

You could say the same about every intrinsic. Do you think we should remove "the" from (say) "the Array constructor", "the Object prototype", and "the global object"? We could probably do more to emphasize the per-Realm-ness of these objects, but I don't think deleting "the" is going to improve the reader's understanding of that point.

Reply:
@jmdyck well all of those are technically needing plural if we don't mention that things are per realm. I think having implicit per-Realm-ness that might show up as a single sentence somewhere in the spec like right now ( https://tc39.github.io/ecma262/#sec-well-known-intrinsic-objects ) just makes the spec harder overall since it is glossing over what is going on. Having an object represent a set of objects as stated in that link is kind of odd.

Reply:
If the problem is implicit per-Realm-ness, then note that removing "the" or pluralizing does nothing to solve it. That is, indefiniteness or plurality don't tell the reader that there's an object like this in every Realm. (E.g., if you talk about "global objects", the reader may think you're talking about globally-accessible objects within the current realm.)

One solution to implicit per-Realm-ness would be to always use wording such as "Within every realm, the global object ..." or "the Array constructor of the realm in question". I think that would get tiresome. It wouldn't be *as* bad if you limited the explicit per-realm reminder to once per clause, but I think it would still be overkill.

Another solution would be to use a special notation (e.g., [the global object]), indicating that the enclosed phrase is only meaningful with respect to some realm. (Explicit insofar as it's a visible marker, but subtle in that it doesn't add any words.) This is *almost* what the %Foo% notation does, except that:
(a) it only 'operates' on the ad-hoc names set out in column 1 of Table 7, and
(b) it only refers to an object in the *current* realm, and the current realm isn't always defined or what you want.

Personally, I'm not convinced that implicit per-Realm-ness is a problem, as long as we make it explicit somewhere. Though, like I say, we could do a better job of this. (E.g., a reminder in clause 17 would probably be a good idea.)

Reply:
(Re "we could do a better job": note issue #532.)

Reply:
@jmdyck we already mention that things are explicitly per realm in the spec, I don't think having it stated multiple times would improve things. A syntax is an interesting idea but I feel like we have enough noise already with `@@`, `[]`, `[[]]` and `%%`. I do think that stating them as a plural would at least give pause to readers that there can be multiple of these things and lead them to find out how multiple of them can exist.

Reply:
I'm going to say this is a non-blocker for me and we can discuss this as a whole spec challenge somewhere else.

Reply:
> we already mention that things are explicitly per realm in the spec, I don't think having it stated multiple times would improve things.

Okay, so that sounds like you agree with me that implicit per-Realm-ness *isn't* a problem. But earlier, you said "I think having implicit per-Realm-ness [...] just makes the spec harder overall" (which is what prompted my suggested 'solutions'), so I'm confused. (But you've approved the PR, so thanks.)

Reply:
It is a problem, just not to be solved here. I think it is harder to understand as it exists today, but probably should be addressed as a separate PR/issue thread.